### PR TITLE
gitnuro: add aarch64-linux support

### PR DESCRIPTION
**Pull request by julius-boettger**
## Description of changes

Add support for `aarch64-linux`, as already discussed [here](https://github.com/NixOS/nixpkgs/pull/278514#issuecomment-2140093673).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by julius-boettger** [comment]
@pbsds we already talked about these changes, could you review this?

**Comment by JohnRTitor** [comment]
@ofborg build gitnuro
@ofborg eval

**Review by JohnRTitor — COMMENTED** [review]
I tried to build and run this, but encountered this error.

<details>

```log
$ nix run github:julius-boettger/nixpkgs/gitnuro-aarch64#legacyPackages.x86_64-linux.pkgsCross.aarch64-multiplatform.gitnuro
error: builder for '/nix/store/7nms8g9l2r9047v1l3rccr6wqp3024ny-gnome-vfs-aarch64-unknown-linux-gnu-2.24.4.drv' failed with exit code 1;
       last 25 log lines:
       > checking sys/mntctl.h presence... no
       > checking for sys/mntctl.h... no
       > checking sys/vfs.h usability... yes
       > checking sys/vfs.h presence... yes
       > checking for sys/vfs.h... yes
       > checking sys/vmount.h usability... no
       > checking sys/vmount.h presence... no
       > checking for sys/vmount.h... no
       > checking sys/sysctl.h usability... no
       > checking sys/sysctl.h presence... no
       > checking for sys/sysctl.h... no
       > checking fstab.h usability... yes
       > checking fstab.h presence... yes
       > checking for fstab.h... yes
       > checking fnmatch.h usability... yes
       > checking fnmatch.h presence... yes
       > checking for fnmatch.h... yes
       > checking util.h usability... no
       > checking util.h presence... no
       > checking for util.h... no
       > checking sys/sysmacros.h usability... yes
       > checking sys/sysmacros.h presence... yes
       > checking for sys/sysmacros.h... yes
       > checking for gconftool-2... no
       > configure: error: gconftool-2 executable not found in your path - should be installed with GConf
       For full logs, run 'nix log /nix/store/7nms8g9l2r9047v1l3rccr6wqp3024ny-gnome-vfs-aarch64-unknown-linux-gnu-2.24.4.drv'.
error: 1 dependencies of derivation '/nix/store/mg18gl1drc65qwbxgppip5jvm128f5dn-openjdk-aarch64-unknown-linux-gnu-21+35.drv' failed to build
error: 1 dependencies of derivation '/nix/store/5vdl6m3c2pcz9glxzv79dzja2k9wz5i3-gitnuro-aarch64-unknown-linux-gnu-1.3.1.drv' failed to build
```

</details>

~~Perhaps it needs `GConf` as build inputs?~~

EDIT: Looks like Ofborg was able to build it successfully.

**Review by JohnRTitor — APPROVED** [review]

File: pkgs/applications/version-management/gitnuro/default.nix [hunk 1]
```diff
@@ -12,10 +12,19 @@ stdenv.mkDerivation rec {
...
+    {
+      url = "https://github.com/JetpackDuba/Gitnuro/releases/download/v${version}/Gitnuro-linux-x86_64-${version}.jar";
+      hash = "sha256-7yne9dD/7VT+H4tIBJvpOf8ksECCpoNAa8TSmFmjYMw=";
+    }
+    else if stdenv.hostPlatform.system == "aarch64-linux" then
+    {
+      url = "https://github.com/JetpackDuba/Gitnuro/releases/download/v${version}/Gitnuro-linux-arm_aarch64-${version}.jar";
+      hash = "sha256-6TRQfIhaKBjNPn3tEVWoUF92JAmwlHUtQZE8gKEZ/ZI=";
+    }
+    else throw "Unsupported architecture"
```
Comment (pbsds) [inline pkgs/applications/version-management/gitnuro/default.nix:26]:
```suggestion
    else throw "Unsupported architecture: ${stdenv.hostPlatform.system}"
```

**Comment by pbsds** [comment]
Please squash the commits

**Comment by julius-boettger** [comment]
Done @pbsds

**Comment by JohnRTitor** [comment]
This and earlier commits should be eligible for stable, could you backport this @julius-boettger?

**Comment by julius-boettger** [comment]
> This and earlier commits should be eligible for stable, could you backport this @julius-boettger?

I have never done that before, can you link me to a guide on how to do that? I also don't have much time right now, so if someone else could do it instead of me, I would also appreciate it.

**Comment by JohnRTitor** [comment]
Backporting involves cherry picking commits from master branch (obviously PRs need to be merged first) into one of supported release branches. Current release is 24.05, so one should clone `release-24.05` branch and make a PR to that branch with the cherry picked changes. 

Guide is [here](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#manually-backporting-changes).

In this particular case, commit from this PR and https://github.com/NixOS/nixpkgs/pull/278514 are to be cherry-picked. 

Automatic backport (can only be done by a maintainer) failed because the earlier change has not been backported yet.

**Pull request merged**